File: frontend/src/services/meals.ts
// src/services/meals.ts
import api from "@/lib/axios";
import { Meal, MealCreateInput } from "@/models/meals";

export const mealService = {
  async getMealsByDate(date: string): Promise<Meal[]> {
    const { data } = await api.get<Meal[]>("/meals", {
      params: { date },
    });
    return data;
  },

  async createMeal(input: MealCreateInput): Promise<Meal> {
    const { data } = await api.post<Meal>("/meals", input);
    return data;
  },

  async deleteMeal(id: string): Promise<{ deleted: boolean }> {
    const { data } = await api.delete<{ deleted: boolean }>(`/meals/${id}`);
    return data;
  },
};

File: backend/app/routers/meals.py
# app/routers/meals.py
from typing import List
from fastapi import APIRouter, Depends, Query
from sqlalchemy.orm import Session

from app.dependencies.db import get_db
from app.schemas.meals import MealCreate, MealOut
from app.services.meals import MealService

router = APIRouter(tags=["meals"])


@router.post("/meals", response_model=MealOut)
def create_meal(payload: MealCreate, db: Session = Depends(get_db)):
    return MealService.create_meal(db, payload)


@router.get("/meals", response_model=List[MealOut])
def get_meals(date: str = Query(..., description="Date in YYYY-MM-DD format"), db: Session = Depends(get_db)):
    return MealService.get_meals_by_date(db, date)


@router.delete("/meals/{meal_id}")
def delete_meal(meal_id: str, db: Session = Depends(get_db)):
    return MealService.delete_meal(db, meal_id)
